lib/HLSL/HLOperationLower.cpp

```diff
@@ -4987,7 +4987,8 @@ Value *TranslateConstBufMatLd(Type *matType, Value *handle, Value *offset,
                               bool colMajor, OP *OP, const DataLayout &DL,
                               IRBuilder<> &Builder) {
   unsigned col, row;
-  Type *EltTy = HLMatrixLower::GetMatrixInfo(matType, col, row);
+  HLMatrixLower::GetMatrixInfo(matType, col, row);
```

Comment:
I would love to follow up with a refactoring that eliminates `GetMatrixInfo` as it currently is because it is important for the callers to know whether they want the register or memory type of the elements, or it will lead to more bool bugs.

In fact, I'd like to have a struct with `getRegElemType`, `getMemElemType`, `getRegVecType`, `getMemVecType`, plus `convertElemRegToMem`, `convertElemMemToReg`, `convertVecRegToMem` and `convertVecMemToReg` helpers.